### DISCUSSION TITLE
Need a reference for existing plasticity codes

### DISCUSSION THREAD

**Discussion by mahyarmalek** [idaholab/moose#19185] [Q&A Modules: Solid mechanics] [answered]
**Hello Moose developers,** 

I'm trying to implement a cam clay model ( a soil plasticity model) within the template of the plasticity codes that already exist in the library. I need to understand how cappeddruckerprager works, and to do so, I need some documentation or references on which this code has been written. 

**Thank you,**

**Mahyar**

**Comment by GiudGiud** [comment, marked as answer]
There's some documentation here
https://mooseframework.inl.gov/source/materials/CappedDruckerPragerCosseratStressUpdate.html

but I had a look at the source code and it s nothing too complicated.
